appcues/src/main/java/com/appcues/data/mapper/MapperExt.kt

```diff
@@ -1,18 +1,16 @@
 package com.appcues.data.mapper
 
 import com.appcues.data.remote.response.action.ActionResponse
-import com.appcues.data.remote.response.trait.TraitResponse
+import com.appcues.trait.ExperienceTrait
+import com.appcues.trait.ExperienceTrait.ExperienceTraitLevel
 import java.util.UUID
 
-internal fun List<TraitResponse>.mergeTraits(other: List<TraitResponse>): List<TraitResponse> {
-    return mutableListOf<TraitResponse>().apply {
+internal fun List<ExperienceTrait>.mergeTraits(others: List<ExperienceTrait>, level: ExperienceTraitLevel): List<ExperienceTrait> {
+    return mutableListOf<ExperienceTrait>().apply {
         addAll(this@mergeTraits)
 
-        other.forEach { trait ->
-            if (none { it.type == trait.type }) {
-                add(trait)
-            }
-        }
+        others.filter { other -> other.level > level }
```

Comment:
for my own education - what is this `other.level > level` check adding now? a comment in here about how this merge works would be nice for future reference.

Reply:
sorry I see you commented on this above, ok - still would be nice to have more explanation in the code I think.

Reply:
whatever you see its confusing, make note so I dont forget to add extra explanation in the code.